src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -221,8 +269,12 @@ void WorkStealingThreadPool::WorkStealingThreadPoolImpl::Quiesce() {
   // running instead of zero.
   bool is_threadpool_thread = g_local_queue != nullptr;
   work_signal()->SignalAll();
-  living_thread_count_.BlockUntilThreadCount(is_threadpool_thread ? 1 : 0,
-                                             "shutting down");
+  auto threads_were_shut_down = living_thread_count_.BlockUntilThreadCount(
+      is_threadpool_thread ? 1 : 0, "shutting down",
+      kBlockUntilThreadCountTimeout);
```

Comment:
Wondering if it's worth preserving the behavior of an infinite timeout if `g_log_verbose_failures == false`.

Otherwise, what happens if we reach the `kBlockUntilThreadCountTimeout` timeout and `g_log_verbose_failures == false` ?

Reply:
Good catch! I agree, fixed.